opennms-webapp/src/main/java/org/opennms/web/admin/discovery/ActionDiscoveryServlet.java

```diff
@@ -206,7 +206,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         //add an 'Include URL'
         if(action.equals(addIncludeUrlAction)){
             LOG.debug("Adding Include URL");
-            String url = request.getParameter("iuurl");
+            String url = WebSecurityUtils.sanitizeString(request.getParameter("iuurl"));
             String timeout = request.getParameter("iutimeout");
             String retries = request.getParameter("iuretries");
             String foreignSource = request.getParameter("iuforeignsource");
```

Comment:
Silly question, but is there a reason foreign source and location shouldn't be sanitized as well? Probably not XSS issue, but could be potentially writing weird things to the filesystem or something.

Reply:
those are drop-downs, not text fields
![image](https://user-images.githubusercontent.com/93448083/184393082-1f73ceff-2e01-41c7-95a7-5855ec9b8225.png)

Reply:
and `Retries` only allows numbers, so nothing bad gets past the UI